core/autoscaling.go

```diff
@@ -566,29 +533,33 @@ func (a *autoScalingGroup) launchCheapestSpotInstance(
 	}
 	logger.Println("Found on-demand instance", *baseInstance.InstanceId)
 
-	newInstanceType, err = a.getInstanceTypeByTagInASG()
-	if err != nil {
-		logger.Println("Errors while trying to fetch tags for instance type...")
-		return errors.New("Errors while trying to fetch tags")
-	}
-
 	// Check option to keep instance type
 	// If we keep the instance type we don't need to calculate the
 	// compatible instance type.
-	// Also ignore if it's already specified in the instance-type tag
-	if newInstanceType == "" {
-		if a.region.conf.KeepInstanceType == "current" {
-			newInstanceType = *baseInstance.InstanceType
-		} else if a.region.conf.KeepInstanceType != "" {
-			newInstanceType = a.region.conf.KeepInstanceType
-		} else {
-			newInstanceType, err = baseInstance.getCheapestCompatibleSpotInstanceType()
-		}
-		if err != nil {
-			logger.Println("No cheaper compatible instance type was found, "+
-				"nothing to do here...", err)
-			return errors.New("no cheaper spot instance found")
-		}
+	// Also ignore if it's already specified in the allowed-instance-types tag
+	allowedInstanceTypesTag := strings.Replace(*a.getTagValue("allowed-instance-types"), " ", ",", -1)
+	allowedInstanceTypes := strings.Replace(a.region.conf.AllowedInstanceTypes, " ", ",", -1)
+
+	// Tag has priority
+	if allowedInstanceTypes == "" {
+		allowed = allowedInstanceTypesTag
+		allowedList = strings.Split(allowed, ", ")
+	} else {
+		allowed = allowedInstanceTypes
+		allowedList = strings.Split(allowed, ", ")
+	}
+
+	if allowed == "current" {
+		newInstanceType = *baseInstance.InstanceType
+	} else if allowed != "" {
+		newInstanceType = allowed
+	} else {
+		newInstanceType, err = baseInstance.getCheapestCompatibleSpotInstanceType(allowedList)
```

Comment:
I would wrap this configuration handling logic inside a function returning the allowedList